homeassistant/components/rest_command/__init__.py

```diff
@@ -198,6 +182,17 @@ async def async_service_handler(service: ServiceCall) -> ServiceResponse:
                 )
                 raise HomeAssistantError from err
 
+            except (JSONDecodeError, AttributeError) as err:
+                _LOGGER.error("Response of `%s` has invalid JSON", request_url)
+                raise HomeAssistantError from err
+
+            except (LookupError, UnicodeDecodeError) as err:
+                _LOGGER.error(
+                    "Response of `%s` could not be interpreted as text",
+                    request_url,
+                )
+                raise HomeAssistantError from err
```

Comment:
This is not a good change, we should catch exceptions where they happen.

Reply:
Reverted catching location to where the exceptions may occur. The slack on the McCabe complexity _just_ allows this.
The linting rules should not be changed in this case as the rules already quite relaxed IMO.

I've also re-introduced the `LookupError` as we're dealing with a `ClientResponse` response object which in contrast to the [`WebRequest`](https://github.com/aio-libs/aiohttp/blob/6f1c608fe756baa4b82ea8714373cebab0252265/aiohttp/web_request.py#L657) does not gracefully handle misrepresented encoding types. 
I.e. when the content type of the response is represented as plain text / json, but contains a binary blob / image data the bytes decode throws a lookup error. 
Definately an edge-case though.

Reply:
I don't read that test case the same way -- isn't the case you describe the `UnicodeDecodeError ` case? I see `LookupError` saying it can't look up the encoding type requested, which won't happen here given we're using a single encoding. (unless i'm missing something or you can author a test that shows how this is possible, i don't think this is a case to worry about. adding a case for trying to unicode binary data would be a reasonable case to add)

Reply:
You're right that the `LookupError` isn't covered in the test case.

I may have been trying to cover too many bases, as it would be an edgecase where the response is malformed. Similar exceptions such as the `ContentTypeError` that may occur in such cases don't seem to be caught elsewhere either.